securedrop_client/api_jobs/sync.py

```diff
@@ -40,7 +42,15 @@ def call_api(self, api_client: API, session: Session) -> Any:
         #
         # This timeout is used for 3 different requests: `get_sources`, `get_all_submissions`, and
         # `get_all_replies`
-        api_client.default_request_timeout = 60
+        api_client.default_request_timeout = int(
+            os.environ.get("SDEXTENDEDTIMEOUT", self.DEFAULT_REQUEST_TIMEOUT)
+        )
+        if api_client.default_request_timeout != self.DEFAULT_REQUEST_TIMEOUT:
+            logger.warn(
+                f"{self.__class__.__name__} will use "
+                f"default_request_timeout={api_client.default_request_timeout}"
+            )
```

Comment:
A minor pet peeve of mine: since the message doesn't require action (e.g. like a deprecation warning does), I would prefer it to be at `INFO` level.

I say pet peeve because I find personally that most warnings shouldn't be emitted because they fall into the following categories:
- Error that are not fatal because they're handled (In my experience, the source of the error is unlikely to be fixed anyway if given the opportunity the choice was made, instead, to print a warning. :shrug: So the warning is mostly noise IMHO.)
- Fatal errors, which shouldn't have happened and would better be logged at the ERROR level since they signal a bug to be fixed.
- Or messages that don't require user action and only provide context. (At which point I think competing for attention at the warning level contributes more to error-fatigue than it really helps keeping us informed. Using an INFO level seems more honest to me.)

What would I make a warning? Things that will become fatal errors unless action is taken, e.g. deprecation warnings. YMMV! Admittedly it's nitpickey and mostly a matter of personal preference. :slightly_smiling_face:

Reply:
I generally agree with your criteria, @gonzalo-bulnes.  In this specific case, I'm rounding up slightly from our decision to call out this flag ["explicitly as experimental, i.e. we cannot guarantee that we will support it in feature"](https://github.com/freedomofpress/securedrop-client/issues/1547#issuecomment-1227633152) to something akin to a deprecation warning.  But I'm happy to demote this message to `INFO` if that logic isn't persuasive.

Reply:
Oh, that makes sense @cfm I missed that intent! I'm not sure what context goes into the current wording: would there be an opportunity to call out explicitly (and additionally to the warning level): "WARNING: ... use experimental default_..."?